### ISSUE TITLE
QEMU provider

### ISSUE THREAD

**Issue by beriberikix** [loft-sh/devpod#335]
Hello! Is it feasible to create a provider for QEMU? I work with embedded software and access to physical devices is critical. Docker and Podman have limitations that make the general case impossible. A full VM like QEMU (and full access to it's features) provides most of what embedded developers need. If I understand the devpod model currently, it should be feasible to create provider for QEMU. If so, is that on the roadmap or something the community has already expressed interest around?

Cheers!

**Comment by FabianKramm** [comment]
@beriberikix thanks for creating this issue! Yes that sounds very feasible and it should be definitely possible to write such a provider, we even thought about creating a kvm provider to use for DevPod ourselves. However, right now, we don't have immediate plans to create such a provider, but would welcome outside contributions.

**Comment by beriberikix** [comment]
Cool! And hyper-v/Hypervisor.framework for Windows/macOS, respectively!

I'd love to find the time to pull something together but I know I won't. On the odd chance I do, is there a starter provider I could look at?

**Comment by FabianKramm** [comment]
@beriberikix yes we have an example provider here https://github.com/loft-sh/devpod/blob/main/examples/simple-k8s-provider/provider.yaml and if you want an actual fully fledged example, you can take a look at https://github.com/loft-sh/devpod-provider-civo

**Comment by koolay** [comment]
Is there any example for `machine provider`?
I want to create a workspace on local qemu vm, but not found any example.

**Comment by FabianKramm** [comment]
@koolay you can also use the ssh provider to connect to it, we can think about adding a qemu provider example to devpod to give you an idea
